creme/creme_core/core/enumerable.py

```diff
@@ -94,8 +95,33 @@ def _queryset(self):
 
         return qs.complex_filter(limit_choices_to) if limit_choices_to else qs
 
-    def choices(self, user):
-        return [*map(self.instance_as_dict, self._queryset())]
+    def filter_by_term(self, user, term, limit=None):
+        queryset = self._queryset()
+        term = term.lower()
+
+        choices = filter(
+            lambda c: term in c['label'].lower(),
+            map(self.instance_as_dict, queryset)
+        )
+
+        return list(islice(choices, limit) if limit else choices)
```

Comment:
I understand why you filter after having retrieved all the instance: you don't know how the label (i.e. `__str__` in general ) is built, so you build all the label then you filter.
But it has 2 big issues:
 - `term in c['label'].lower()` does not work well with diacritics I presume. It could be done in Python, so it's not the main issue (but I can give idea of sadistic unit tests...).
 - the filtering is not made by the SQL server. The returned JSON will be short (so it avoid a part of the performance issue), but it's clearly not satisfactory.

So we must found which field(s) should be used in order to make a '__icontains' query.
 - 1) If the related model gets a "_search_fields" attribute, use them.
 - 2) If not, search the first viewable CharField in the model (404/409/400 if no field is found of course)

I presume that our models could remain the same (i.e. not need to add the attribute "_search_fields") : the rule 2 will work perfectly, our model registered in CremeConfig have just a CharField title/name in 90% of the cases.
We prefix "_search_fields" with an underscore because it's clearly an "API" which we will break for a more powerful system in the future.
Notice that for problematic modes, people can still register their own Enumerator.